packages/backend/src/apps/formsg/common/form-env.ts

```diff
@@ -0,0 +1,84 @@
+import type { IGlobalVariable } from '@plumber/types'
+
+import formsgSdk from '@opengovsg/formsg-sdk'
+
+import { FORM_ID_LENGTH } from './constants'
+
+const STAGING_SDK = formsgSdk({
+  mode: 'staging',
+})
+const PRODUCTION_SDK = formsgSdk({
+  mode: 'production',
+})
+
+// TODO: add UAT env
+export const SUPPORTED_FORM_ENVS = ['prod', 'staging'] as const
+export type FormEnv = (typeof SUPPORTED_FORM_ENVS)[number]
+
+export function parseFormIdAsUrl(rawUrl: string): URL | null {
+  try {
+    const url = new URL(
+      rawUrl.startsWith('http') ? rawUrl : `https://${rawUrl}`,
+    )
+
+    return url.hostname === 'form.gov.sg' ||
```

Comment:
can i check why is the first condition check necessary ah? doesn't the 2nd one alr cover the first

Reply:
2nd check has the `.` in front, which is needed cos we dont wanna match deform.gov.sg